components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/jwt/JWTValidator.java

```diff
@@ -176,6 +183,42 @@ public AuthenticationContext authenticate(SignedJWTInfo signedJWTInfo, MessageCo
                 throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
                         "Invalid JWT token");
             }
+            if (jwtGeneratedTime != 0 && InternalRevokedJWTDataHolder.getInstance()
+                    .isJWTTokenClientIdExistsInRevokedMap((String) signedJWTInfo.getJwtClaimsSet()
+                            .getClaim("client_id"), jwtGeneratedTime)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Consumer key retrieved from the  jwt token map is in revoked consumer key map." +
+                            " Token: " + GatewayUtils.getMaskedToken(jwtHeader));
+                }
+                log.error("Invalid JWT token. " + GatewayUtils.getMaskedToken(jwtHeader));
+                throw new APISecurityException(APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
+                        "Invalid JWT token");
+            }
+            if (jwtGeneratedTime != 0 && signedJWTInfo.getJwtClaimsSet().getSubject()
+                    .equals(signedJWTInfo.getJwtClaimsSet().getClaim("client_id"))
```

Comment:
Better to use a string constant for "client_id" here.

Reply:
Fixed by. https://github.com/wso2/carbon-apimgt/pull/12174/commits/70b076a6067da5daa54577ca7065398e9a9c5e7d